hibernate-core/src/main/java/org/hibernate/query/hql/internal/SemanticQueryBuilder.java

```diff
@@ -1283,28 +1283,89 @@ else if ( ctx.GREATER_EQUAL()!=null ) {
 
 	@Override
 	public SqmComparisonPredicate visitComparisonPredicate(HqlParser.ComparisonPredicateContext ctx) {
+		ComparisonOperator comparisonOperator = (ComparisonOperator) ctx.comparisonOperator().accept( this );
```

Comment:
Few things in this block to keep in mind...

1. We prefer to define final variables as `final`.  You have quite a few here.
2. We prefer code blocks use braces, including switch branches (i.e., `case NOT_EQUAL: { ... }`)

Reply:
Thanks I'll keep this in mind.